lib/ukstore/store.c

```diff
@@ -0,0 +1,241 @@
+
+	uk_list_add(&folder->folder_head, &dynamic_heads[library_id]);
+
+	return 0;
+}
+
+/**
+ * Searches for an folder in a library and returns it.
+ *
+ * @param library_id the library id to search in
+ * @param name the name of the folder to search for
+ * @return the found folder or NULL
+ */
+struct uk_store_folder *
+_uk_store_get_dynamic_folder(unsigned int library_id, const char *name)
+{
+	struct uk_store_folder *res = NULL;
+
+	if (!dynamic_heads[library_id].next)
+		return NULL;
+
+	uk_list_for_each_entry(res, &dynamic_heads[library_id], folder_head)
+		if (!strcmp(res->folder_name, name))
+			return res;
+
+	return NULL;
+}
+
+/**
+ * Searches for an folder in a library and returns it.
+ *
+ * @param library_id the library id to search in
+ * @param name the name of the folder to search for
+ * @return the found folder or NULL
+ */
+struct uk_store_folder *
+_uk_store_get_static_folder(unsigned int library_id, const char *name)
+{
+	struct uk_store_folder *sec_start = static_folders[2 * library_id];
+	struct uk_store_folder *sec_end = static_folders[2 * library_id + 1];
+
+	if (!sec_start)
+		return NULL;
+
+	for (int i = 0; &sec_start[i] != sec_end; ++i)
+		if (!strcmp(sec_start[i].folder_name, name))
+			return &sec_start[i];
+
+	return NULL;
+}
+
+/**
+ * Searches for an entry in a folder and returns it. Increases the refcount.
+ *
+ * @param folder the folder to search in
+ * @param name the name of the entry to search for
+ * @return the found entry or NULL
+ */
+struct uk_store_entry *
+uk_store_get_entry(struct uk_store_folder *folder, const char *name)
+{
+	struct uk_store_folder_entry *res = NULL;
+
+	uk_list_for_each_entry(res, &folder->entry_head, list_head)
+		if (!strcmp(res->entry->entry_name, name))
+			break;
+
+	if (res) {
+		ukarch_inc(&res->refcount.counter);
+		return res->entry;
+	}
+
+	return NULL;
+}
+
+/**
+ * Calls the saved setter with the value in `in`
+ *
+ * @param entry the entry to call from
+ * @param in the value to give to the setter
+ */
+int
+uk_store_set_value(struct uk_store_entry *entry, void *in)
+{
+	if (unlikely(!entry))
+		return -EINVAL;
+
+	switch (entry->type) {
+	case uptr:
+		entry->set.uptr(*((__uptr *) in));
+		break;
+
+	case s64:
+		entry->set.s64(*((__s64 *) in));
+		break;
+
+	case u64:
+		entry->set.u64(*((__u64 *) in));
+		break;
+
+	case s32:
+		entry->set.s32(*((__s32 *) in));
+		break;
+
+	case u32:
+		entry->set.u32(*((__u32 *) in));
+		break;
+
+	case s16:
+		entry->set.s16(*((__s16 *) in));
+		break;
+
+	case u16:
+		entry->set.u16(*((__u16 *) in));
+		break;
+
+	case s8:
+		entry->set.s8(*((__s8 *) in));
+		break;
+
+	case u8:
+		entry->set.u8(*((__u8 *) in));
+		break;
+
+	default:
+		return -EINVAL;
+	}
+	return 0;
+}
+
+/**
+ * Calls the saved getter and saves the returned value in out
+ *
+ * @param entry the entry to call from
+ * @param out the value to return
+ */
+int
+uk_store_get_value(struct uk_store_entry *entry, void *out)
```

Comment:
I would do the something similar for the getters. Also here you can check if the returned number is still in range:
``` c
int uk_store_get_u8(struct uk_store_entry *e, __u8 *out);
int uk_store_get_u16(struct uk_store_entry *e, __u16 *out);
int uk_store_get_u32(struct uk_store_entry *e, __u32 *out);
int uk_store_get_charp(struct uk_store_entry *e, char *buf, size_t buflen); /* uses snprintf() for number entries */
```

Reply:
Same thing here like the setters.